app/controllers/articles_controller.rb

```diff
@@ -276,8 +276,8 @@ def handle_notifications(updated)
     if updated && @article.published && @article.saved_changes["published"] == [false, true]
       Notification.send_to_followers(@article, "Published")
     elsif @article.saved_changes["published"] == [true, false]
-      Notification.remove_all_without_delay(notifiable_id: @article.id, notifiable_type: "Article", action: "Published")
-      Notification.remove_all(notifiable_id: @article.id, notifiable_type: "Article", action: "Reaction")
+      Notification.remove_all_by_action_without_delay(notifiable: @article, action: "Published")
+      Notification.remove_all(@article.comments.to_a)
```

Comment:
This could be a lot of items...

Reply:
There will also be no need to create an ActiveRecord objects collection which can be huge, if you pass only ids and type.